0_web_functions.R

```diff
@@ -114,12 +114,12 @@ get_input_files <- function(portfolio_name_ref_all) {
   # set_portfolio_parameters(file_path = file.path(par_file_path, "PortfolioParameters.yml"))
   # check that the provided reference names match to the input files
 
-  input_files <- list.files(path = input_path, full.names = F)
-  input_names <- gsub(".csv", "", input_files)
-  input_names <- gsub(".txt", "", input_names)
-  input_names <- gsub(".xlsx", "", input_names)
+  input_files <- list.files(path = input_path, full.names = FALSE)
+  input_names <- tools:::file_path_sans_ext(input_files)
+  portfolio_files <- grep(portfolio_name_ref_all, input_files,
+                          value = TRUE, fixed = TRUE)
+  input_file_type <- unique(tools::file_ext(portfolio_files))
```

Comment:
I guess this will cause an error if a user uploads two different types of files right? I'm not sure if it's possible and what we should do about this though..

Reply:
this PR should have the exact same behavior as before except for specifically treating `portfolio_name_ref_all` as a fixed string rather than a regex... other changes here were cosmetic/stylistic

but... it will likely give a warning (though not an error) on L123 if there is more than one file type because only the first element in what could be a multi-element result will be evaluated in the `if` statement.

however, as far as I can tell, that whole chunk testing the file extensions is irrelevant since L166 forces the filename to end in `.csv` anyway... so apparently this function is only able to read in CSV files, otherwise it will fail because it forces the file extension to be `.csv`